### PR TITLE
Adds an Artifact creation interface

### PR DESCRIPTION
**Pull request by jakereps**
- [x] Add an Artifact creation UI
    - [x] Add action functionality
- [x] Add server interaction to create Artifacts

### CODE REVIEW

File: app/js/actions/artifacts.js [hunk 1]
```diff
@@ -25,6 +25,18 @@ export const clearArtifacts = () => ({
...
 
+export const createArtifact = (dirPath, type) => {
+    return (dispatch, getState) => {
+        const { connection: { uri, secretKey } } = getState();
+        const body = {
+            path: dirPath,
+            type
+        };
+        fetchAPI(secretKey, 'POST', `http://${uri}/api/workspace/artifacts`, body)
+        .then(() => dispatch(actions.refreshArtifacts()));
```
Comment (ebolyen) [inline app/js/actions/artifacts.js]: We'll want a `catch` probably. See `auth.js` for example.

File: app/js/components/ArtifactGenerator.jsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                    <input
+                        type="text"
+                        disabled="true"
+                        className="form-control"
+                        style={{ width: '100%' }}
+                        placeholder="Data Directory Path"
+                        value={dirPath || ''}
+                    />
+                </div>
+            </td>
```
Comment (ebolyen) [inline app/js/components/ArtifactGenerator.jsx]: We'll want a name key in here somewhere.

File: app/js/actions/currentdirectory.js [hunk 3]
```diff
@@ -35,3 +35,24 @@ export const directoryChangeDialog = (currPath) => {
...
+});
+
+export const selectArtifactDirectory = () => {
+    return (dispatch, getState) => {
+        const currPath = getState().currentDirectory;
+        remote.dialog.showOpenDialog({
+            title: 'Choose Artifact Directory',
+            defaultpath: currPath,
+            buttonlabel: 'Set Artifact Directory',
+            properties: ['openDirectory']
```
Comment (ebolyen) [inline app/js/actions/currentdirectory.js]: The API will also accept a filepath it turns out.

File: app/js/components/ArtifactGenerator.jsx (hunk 2 above)
Comment (jakereps) [inline app/js/components/ArtifactGenerator.jsx]: This isn't actually inside of the `<form>`, so it's never being looked up or referenced. The path is being taken out of state and that `dirPath` variable is just being passed to the `onSubmit` function. So, it's basically just for looks and to show the user what they've selected.

File: qiime_studio/api/workspace.py [hunk 4]
```diff
@@ -63,6 +63,18 @@ def get_artifacts():
     return jsonify({'artifacts': artifacts})
 
 
+@workspace.route('/artifacts', methods=['POST'])
+def create_artifact():
+    try:
+        request_body = request.get_json()
+        Artifact.import_data(request_body['path'], request_body['type'])
```
Comment (ebolyen) [inline qiime_studio/api/workspace.py]: The order of the params is about to flip on this.

**Comment by jakereps** [comment]
Should be good to go! Thanks, @ebolyen!

**Comment by ebolyen** [comment]
Thanks @jakereps!

**Pull request merged**